### DISCUSSION TITLE
C API for encoding as tones?

### DISCUSSION THREAD

**Discussion by kubukoz** [ggerganov/ggwave#105] [Q&A]
Hi! I'm liking the library a lot so far, but when I tried using it on an embedded device (a [Playdate](https://play.date/)), it runs pretty slow with the usual TX encoding. I tried switching the mode to `GGWAVE_OPERATING_MODE_TX_ONLY_TONES`, but it just logs

`Tx is disabled - cannot transmit data with this GGWave instance`

and returns 0. Looking at the code in the `ggwave_encode` function of the C API, it seems to look directly at `ggWave->txWaveform`, ignoring tones. Also, it calls `GGWave::init` explicitly, which logs the above message if `m_isTxEnabled` is false (which it is, in `TX_ONLY_TONES` mode).

So... I don't really know how one would encode as "only tones", even more so with just the C API. Am I missing something?

**Comment by ggerganov** [comment]
Thanks for the interest in `ggwave`!

The [r2t2](https://github.com/ggerganov/ggwave/tree/master/examples/r2t2) and [arduino-tx](https://github.com/ggerganov/ggwave/blob/master/examples/arduino-tx/arduino-tx.ino) examples demonstrate the usage of "only tones" mode:

https://github.com/ggerganov/ggwave/blob/bac95c8113c9bfa25180d9f38f84306132cb1312/examples/r2t2/main.cpp#L73-L75
